diff --git a/make.py b/make.py
index eb275af..5e683c3 100755
--- a/make.py
+++ b/make.py
@@ -357,6 +357,32 @@ class ECPIX5(Board):
             "sdcard",
         }, bitstream_ext=".svf")
 
+# Colorlight i5 support ------------------------------------------------------------------------------------
+
+class Colorlight_i5(Board):
+    SPIFLASH_PAGE_SIZE    = 256
+    SPIFLASH_SECTOR_SIZE  = 4*kB
+    SPIFLASH_DUMMY_CYCLES = 8
+    soc_kwargs = {
+        "sys_clk_freq": int(50e6),     # 48MHz default.
+        "integrated_rom_size": 0xa000, # Reduce integrated_rom_size.
+        "l2_size"    :  2048, # Use Wishbone and L2 for memory accesses.
+        "uart_baudrate": 115.2e3,
+    }
+    def __init__(self):
+        from litex_boards.targets import colorlight_i5
+        from litex.build.tools import write_to_file
+        Board.__init__(self, colorlight_i5.BaseSoC, soc_capabilities={
+            # Communication
+            "serial",
+            "ethernet",
+        }, bitstream_ext=".svf")
+        tdir = os.path.join("build/colorlight_i5/software", "include/generated")
+        os.makedirs(tdir, exist_ok=True)
+        write_to_file(os.path.join(tdir, "target.h"),
+            "// Colorlight i5 needs this to disable TX data to clock delay.\n"
+	    "#define TARGET_BIOS_INIT_FUNC() mdio_write(0, 0x1c, 0x8c00)")
+
 #---------------------------------------------------------------------------------------------------
 # Intel Boards
 #---------------------------------------------------------------------------------------------------
@@ -450,6 +476,7 @@ supported_boards = {
     "camlink_4k":   CamLink4K,
     "trellisboard": TrellisBoard,
     "ecpix5":       ECPIX5,
+    "colorlight_i5":Colorlight_i5,
 
     # Altera/Intel
     "de0nano":      De0Nano,
