diff --git a/drivers/mmc/host/litex_mmc.c b/drivers/mmc/host/litex_mmc.c
index 9bdc088dd100..3b6b0b127aca 100644
--- a/drivers/mmc/host/litex_mmc.c
+++ b/drivers/mmc/host/litex_mmc.c
@@ -281,6 +281,13 @@ static void litex_request(struct mmc_host *mmc, struct mmc_request *mrq)
 
 			sg_copy_to_buffer(data->sg, data->sg_len,
 					host->buffer, write_length);
+#if 1
+			asm(".word 0x500F");
+			unsigned int i;
+			for(i=0;i<2*2048/4;i++) {
+				((volatile unsigned int *) 0xc0000000)[i];
+			}
+#endif
 
 			litex_write8(host->sdwriter +
 					 LITEX_MMC_SDMEM2BLK_ENA_OFF, 0);
@@ -343,6 +350,13 @@ static void litex_request(struct mmc_host *mmc, struct mmc_request *mrq)
 		data->bytes_xfered = min(data->blksz * data->blocks,
 					mmc->max_req_size);
 		if (transfer == SDCARD_CTRL_DATA_TRANSFER_READ) {
+#if 1
+			asm(".word 0x500F");
+			unsigned int i;
+			for(i=0;i<2*2048/4;i++) {
+				((volatile unsigned int *) 0xc0000000)[i];
+			}
+#endif
 			sg_copy_from_buffer(data->sg, sg_nents(data->sg),
 				host->buffer, data->bytes_xfered);
 		}
